addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/dto/HueStateColorBulb.java

```diff
@@ -78,17 +78,65 @@ public HueStateColorBulb(HSBType hsb) {
      * Converts this HueState to a HSBType
      */
     public HSBType toHSBType() {
-        int bri = this.bri * 100 / MAX_BRI;
-        int sat = this.sat * 100 / MAX_SAT;
-        int hue = this.hue * 360 / MAX_HUE;
-
-        if (!this.on) {
-            bri = 0;
-        }
-        if (colormode == ColorMode.ct) {
-            sat = 0;
+        if (colormode == ColorMode.xy) {
+            int i;
+            double d = this.xy[0];
+            d = this.xy[1];
+            double Y = ((double) this.bri) / 100.0d;
+            double X = (Y / this.xy[1]) * this.xy[0];
+            double Z = (Y / this.xy[1]) * ((1.0d - this.xy[0]) - this.xy[1]);
+            int R = (int) (Math.abs(((1.4628067016601562d * X) - (0.18406230211257935d * Y)) - (0.2743605971336365d * Z)) * 255.0d);
```

Comment:
I prefer to stick to standard naming or do we break the algorithm then?

Reply:
What do you mean by "standard naming"? X, Y and Z are really the names for this (Z being the brightness but we're in a 3 dimensional colorspace here).

Reply:
sorry I meant local variables implies using lower case :-)

Reply:
Ah, that would be possible of course. I'm not aware of anything that might break by doing that.